pkg/build/cleanup.go

```diff
@@ -11,8 +11,16 @@ import (
 // temporary directories created by STI build and it also cleans the temporary
 // Docker images produced by LayeredBuild
 type DefaultCleaner struct {
-	util.FileSystem
-	docker.Docker
+	fs     util.FileSystem
+	docker docker.Docker
```

Comment:
Sorry yes, I should have said interface and not struct, but the point I was trying to make is that if you have a function that takes Interface I, and you want to pass S as an instance of I, and T implements I, then that is a good reason to embed T (over including it as a norma field in S).  Otherwise all you're doing is saving some typing in exchange for the dubious benefit of exposing all the behavior of T at the top level of S, which as I noted in your Counter example, can be undesirable.